### PR TITLE
[Gardening] fix looking odd logical expression mixing bool and enum

### PR DESCRIPTION
**Pull request by valeriyvan**
Fix looking odd logical expression mixing bool and enum.

### CODE REVIEW

**Review by atrick — APPROVED** [review]
It was intentional that IsInterestingUser works in place of a boolean but still isn't quite proper to do so.

**Comment by valeriyvan** [comment]
ping

File: lib/SILOptimizer/Utils/CanonicalizeOSSALifetime.cpp [hunk 1]
```diff
@@ -400,8 +400,7 @@ void CanonicalizeOSSALifetime::extendLivenessThroughOverlappingAccess() {
           findLastConsume = !destroys.contains(&inst);
           continue;
         }
-        // Stop at the latest use. An earlier end_access does not overlap.
```
[the remark this answers is not in the document]
Comment (valeriyvan) [reply lib/SILOptimizer/Utils/CanonicalizeOSSALifetime.cpp:403]: It wasn't intentional. Restored comment.

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test and merge

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test Linux

**Pull request merged**